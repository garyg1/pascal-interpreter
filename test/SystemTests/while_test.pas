program while_test;

var x : integer = 1;
var did_break : boolean = false;
begin
    while x < 1000 do
    begin
        if x = 32 then
        begin
            x := x + 1;
            writeln('x = ', x, ', if was true');
            continue;
        end;

        x := x * 2;
        if x > 1000 then
            while x > 1050 do
            begin
                x := x - 1;
                writeln('while, ', x);
                if (x = 1052) then
                begin
                    did_break := true;
                    break;
                end;
            end;

        if did_break then
            break;

        writeln('x = ', x, ', didn''t break');
    end;
end.