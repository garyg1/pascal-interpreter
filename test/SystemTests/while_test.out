x = 2, didn''t break
x = 4, didn''t break
x = 8, didn''t break
x = 16, didn''t break
x = 32, didn''t break
x = 33, if was true
x = 66, didn''t break
x = 132, didn''t break
x = 264, didn''t break
x = 528, didn''t break
while, 1055
while, 1054
while, 1053
while, 1052
